shoop/testing/factories.py

```diff
@@ -206,9 +206,11 @@ def post(self, create, extracted, **kwargs):
             sp.categories = shop.categories.all()
 
 
-def get_address(**overrides):
+def get_address(immutable=False, **overrides):
     data = dict(DEFAULT_ADDRESS_DATA, **overrides)
-    return Address(**data)
+    if immutable:
+        return ImmutableAddress.objects.create(**data)
+    return MutableAddress.objects.create(**data)
```

Comment:
Now that I think about this. It's probably best to just drop the `Address.from_data` and add it as two separate class methods: `MutableAddress.from_data` and `ImmutableAddress.from_data`. That would make it less confusing.

```python
class ImmutableAddress(...):
    @classmethod
    def from_data(cls, data):
        """
        Create immutable address from a data dictionary.

        :param data: data for address
        :type data: dict[str,str]
        :return: Saved immutable address
        :rtype: Address
        """
        # Populate all known address fields even if not originally in `data`
        data_with_all_fields = get_data_dict(cls(**data))
        address = cls.objects.filter(**data_with_all_fields).first()
        return address if address else cls.objects.create(**data_with_all_fields)


class MutableAddress(...):
    @classmethod
    def from_data(cls, data):
        """
        Construct mutable address from a data dictionary.

        :param data: data for address
        :type data: dict[str,str]
        :return: Unsaved mutable address
        :rtype: MutableAddress
        """
        return cls(**data)
```
Then this `get_address` would use those...

Would this be clearer in your opinion?